src/gui/systray.cpp

```diff
@@ -133,123 +124,220 @@ void Systray::setToolTip(const QString &tip)
-    if (QGuiApplication::screens().count() > 1) {
-        trayScreen = QGuiApplication::screens().at(screenIndex());
+void Systray::pauseResumeSync()
+{
+    if (_syncIsPaused) {
+        _syncIsPaused = false;
+        emit resumeSync();
     } else {
-        trayScreen = QGuiApplication::primaryScreen();
+        _syncIsPaused = true;
+        emit pauseSync();
     }
+}
 
-    int screenWidth = trayScreen->geometry().width();
-    int screenHeight = trayScreen->geometry().height();
-    int availableWidth = trayScreen->availableGeometry().width();
-    int availableHeight = trayScreen->availableGeometry().height();
-
-    QPoint topRightDpiAware = QPoint();
-    QPoint topLeftDpiAware = QPoint();
-    if (this->geometry().left() == 0 || this->geometry().top() == 0) {
-        // tray geometry is invalid - QT bug on some linux desktop environments
-        // Use mouse position instead. Cringy, but should work for now
-        topRightDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = QCursor::pos() / trayScreen->devicePixelRatio();
-    } else {
-        topRightDpiAware = this->geometry().topRight() / trayScreen->devicePixelRatio();
-        topLeftDpiAware = this->geometry().topLeft() / trayScreen->devicePixelRatio();
-    }
+/********************************************************************************************/
+/* Helper functions for cross-platform tray icon position and taskbar orientation detection */
+/********************************************************************************************/
 
-    // get x coordinate from top center point of tray icon
-    int trayIconTopCenterX = (topRightDpiAware - ((topRightDpiAware - topLeftDpiAware) * 0.5)).x();
+QScreen *Systray::currentScreen() const
+{
+    const auto screens = QGuiApplication::screens();
+    const auto cursorPos = QCursor::pos();
+    const auto it = std::find_if(screens.cbegin(), screens.cend(), [=](QScreen *screen) {
+        return screen->geometry().contains(cursorPos);
+    });
+    return (it != screens.cend()) ? *it : nullptr;
+}
 
-    if (availableHeight < screenHeight) {
-        // taskbar is on top or bottom
-        if (trayIconTopCenterX + (400 * 0.5) > availableWidth) {
-            return availableWidth - 400 - 12;
-        } else {
-            return trayIconTopCenterX - (400 * 0.5);
-        }
-    } else {
-        if (trayScreen->availableGeometry().x() > trayScreen->geometry().x()) {
-            // on the left
-            return (screenWidth - availableWidth) + 6;
-        } else {
-            // on the right
-            return screenWidth - 400 - (screenWidth - availableWidth) - 6;
-        }
+QVariant Systray::currentScreenVar() const
+{
+    const auto screens = QGuiApplication::screens();
+    const auto curScreen = currentScreen();
+
+    for(auto i = 0; i < screens.size(); i++) {
+        if(screens[i] == curScreen)
+            return i;
     }
-#endif
+
+    return 0;
```

Comment:
Of course if we use indexOf() we should make sure we don't return -1 since that's what indexOf would give us if curScreen isn't in screens for some (weird) reason.